tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py

```diff
@@ -1,218 +1,268 @@
-import unittest
-
 import numpy
 import six
 
 import chainer
-from chainer.backends import cuda
 from chainer.functions.normalization import batch_renormalization
-from chainer import gradient_check
 from chainer import testing
-from chainer.testing import attr
-from chainer.testing import condition
-
-
-def _batch_renormalization(expander, gamma, beta, x, mean, var, r, d):
-    mean = mean[expander]
-    std = numpy.sqrt(var)[expander]
-    y_expect = (gamma[expander] * ((x - mean) / std * r + d) + beta[expander])
-    return y_expect
 
 
 # naive implementation of differentiable batch renormalization
 def _naive_batch_renormalization(
-        x, gamma, beta, rmax, dmax, eps, avg_mean, avg_std, axis):
-    shape = x.shape
-    stat_shape = list(shape)
-    for i in axis:
-        stat_shape[i] = 1
-    stat_shape = tuple(stat_shape)
-    gamma = chainer.functions.reshape(gamma, stat_shape)
-    beta = chainer.functions.reshape(beta, stat_shape)
-    avg_mean = avg_mean.reshape(stat_shape)
-    avg_std = avg_std.reshape(stat_shape)
-
-    mean = chainer.functions.mean(x, axis=axis, keepdims=True)
-    std = chainer.functions.sqrt(
-        eps +
-        chainer.functions.mean(
-            chainer.functions.square(x - mean),
-            axis=axis, keepdims=True))
-    r = (std.array / avg_std).clip(1./rmax, rmax)
-    d = ((mean.array - avg_mean) / avg_std).clip(-dmax, dmax)
-    xhat = ((x - mean) / std) * r + d
-    return gamma * xhat + beta
-
-
-@testing.parameterize(*(testing.product({
-    'ndim': [0, 1, 2],
-    'eps': [2e-5, 1e-1],
-    'dtype': [numpy.float32],
-    'update_statistics': [True, False],
-}) + testing.product({
-    'ndim': [1],
-    'eps': [2e-5, 1e-1],
-    'dtype': [numpy.float16, numpy.float32, numpy.float64],
-    'update_statistics': [True, False],
-})))
-class TestBatchRenormalization(unittest.TestCase):
+        x, gamma, beta,  # variables
+        mean, var,  # variables
+        running_mean, running_var,  # arrays
+        rmax, dmax, eps, decay):
+    # If decay is not None, the running stats are updated.
+
+    F = chainer.functions
```

Comment:
I don't think this declaration is needed since "F" is only used once within this scope.

Reply:
It's for readability. The number of occurrences should not matter.

Reply:
It's a detail but this way of declaring "F" is rather uncommon and it's declaration is several lines aways from its usage which doesn't really speak for readability IMHO.

Reply:
- To me, `F` is so common that it's readable in Chainer.
- Using `F.batch_renormalization` and `F.sqrt` with `import chainer.functions as F` is another option because there's no issue with circular dependency in the test.

Reply:
Let me merge it as is. I think it's mostly a cosmetic matter. I agree though that it'd be fine or maybe cleaner to declare "F" at the top.